includes/Checker/Checks.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Checker\Check_Context;
+use ReflectionClass;
+use Exception;
+
+/**
+ * Class to run checks on a plugin.
+ *
+ * @since n.e.x.t
+ */
+class Checks {
+
+	/**
+	 * Context for the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 * @var Check_Context
+	 */
+	protected $check_context;
+
+	/**
+	 * Sets the main context and the main file of the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_main_file Absolute path to the plugin main file.
+	 */
+	public function __construct( $plugin_main_file ) {
+		$this->check_context = new Check_Context( $plugin_main_file );
+	}
+
+	/**
+	 * Runs a single check against the plugin.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $checks Ab array of Check class names to run.
+	 * @return Check_Result Object containing all check results.
+	 *
+	 * @throws Exception Thrown when check fails with critical error.
+	 */
+	public function run_checks( array $checks ) {
+		$result     = new Check_Result( $this->check_context );
+		$all_checks = $this->get_checks();
+
+		// Create an array of Check objects to run based on the check names passed.
+		$checks_to_run = array_filter(
+			$all_checks,
+			function( $check ) use ( $checks ) {
+				$check_name = ( new ReflectionClass( $check ) )->getShortName();
+				return in_array( $check_name, $checks, true );
+			}
+		);
```

Comment:
I realize we had not defined that before, but this is a bit overcomplicated. Not a fan of using `ReflectionClass` either.

We will need check orchestration to happen outside this class, in the runner classes probably, so that means those outside classes need access to the check instances anyway. I think we should simply expect an array of `Check` instances for this method. Then the only validation we would need to do is whether the given checks are part of the `$all_checks` array (by instance comparison).